web/src/util.js

```diff
@@ -1,5 +1,254 @@
-const example = x => x
+import { useState } from "react"
 
-export {
-    example
-}
\ No newline at end of file
+const headerContentTypes = [
+    "application/json",
+    "application/xml",
+    "application/javascript",
+    "application/pdf",
+    "application/zip",
+    "application/octet-stream",
+    "application/ld+json",
+    "application/vnd.api+json",
+    "application/msword",
+    "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+    "application/vnd.ms-excel",
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    "application/vnd.ms-powerpoint",
+    "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+    "application/x-www-form-urlencoded",
+    "application/x-tar",
+    "application/x-rar-compressed",
+    "application/x-7z-compressed",
+    "audio/mpeg",
+    "audio/wav",
+    "image/jpeg",
+    "image/png",
+    "image/gif",
+    "image/webp",
+    "image/svg+xml",
+    "multipart/form-data",
+    "text/html",
+    "text/plain",
+    "text/css",
+    "text/javascript",
+    "text/xml",
+    "text/csv",
+    "text/markdown",
+    "video/mp4",
+    "video/mpeg",
+    "video/webm",
+    "video/ogg"
+]
+
+const emptyValue = "<Enter Value>"
+
+const highlightPartOfString = (found, string) => {
```

Comment:
Für die Funktion brauche ich einen besseren Namen

Reply:
mmm...okay, du gehst immer nur von einem match aus, aber tatsächlich kann der Suchbegriff ja auch an mehreren Stellen gefunden werden und wir wollen alle highlighten. Da kommst du dann mit den highlighted-Parts nicht weiter. Um das zu lösen haben wir 2 Ansätze:
- du erzeugst einen String der die matches mit `<span ...>` und `</span>`umschliesst und kannst diesen String in einer State-Variablen speichern und über `dangerouslySetInnerHTML()` in der Option ausgeben
- du schreibst eine Komponente, welche das übernimmt, etwa so:
```
function HighlightMatches({ text, search, className }) {
    const parts = splitByMatch(text, search)
    return (
        <span>
            parts.map(part =>
                isMatch(part) ?
                   (<span className={className}>{part}</span>) :
                   (part)
            )
        </span>
    )
}
```

Reply:
Meinst du mit mehreren stellen, dass in einem String, zwei matches sein können? 

zb sowas wie "AppblablaApp" sodass App an beiden stellen markiert werden soll? Oder verstehe ich das falsch?

Reply:
ja genau, alles was auf die eingabe passt highlighten